Stratis.Bitcoin.Features.Consensus/PosConsensusValidator.cs

```diff
@@ -312,6 +312,17 @@ public override void ContextualCheckBlockHeader(ContextInformation context)
                 ConsensusErrors.BlockTimestampTooEarly.Throw();
             }
 
+            int lastCheckpointHeight = this.checkpoints.GetLastCheckpointHeight();
```

Comment:
Ah and you can fairly easily write a test to check that.

Reply:
I've already created an issue for such a test - https://github.com/stratisproject/StratisBitcoinFullNode/issues/493 - and frankly, once this test exists, I should be able to find a place where to but the check. I was expecting what you just confirmed, so I need to rewrite this. If you could help with the test, that would help a lot.

I'm not sure that this is attacker only scenario though. A poorly connected big staker (say 40 % of the network) will have some chance to achieve this by mistake. Well it is a pretty big mistake, but I would argue that it is possible without any malicious code being present on that particular node. I would the normal consensus temporary ban as it is usual is the right reaction. But since we don't implement these bans yet, this is to be decided later anyway.

OK, so to progress here - if you could help with the test, this will really help me in getting that extra bit of knowledge that is needed to find the spot. And certainly I would then start with those places you've mentioned.

Reply:
Unfortunately we don't have many POS tests yet until I remove the static flags from NBitcoin, but such a test does not depend on POS I can simulate this using POW.

Yes sure I will create a test with 510 size reorg and submit a PR.